bundle/bundle.go

```diff
@@ -161,14 +140,74 @@ func Read(r io.Reader) (Bundle, error) {
 
 		} else if strings.HasSuffix(path, manifestExt) {
 			if err := util.NewJSONDecoder(&buf).Decode(&bundle.Manifest); err != nil {
-				return bundle, errors.Wrapf(err, "bundle load failed on manifest")
+				return bundle, errors.Wrap(err, "bundle load failed on manifest decode")
+			}
+
+			if r.includeManifestInData {
+				var metadata map[string]interface{}
+				b, err := json.Marshal(&bundle.Manifest)
+				if err != nil {
+					return bundle, errors.Wrap(err, "bundle load failed on manifest marshal")
+				}
+
+				err = util.UnmarshalJSON(b, &metadata)
+				if err != nil {
+					return bundle, errors.Wrap(err, "bundle load failed on manifest unmarshal")
+				}
+
+				if err := bundle.insert(manifestPath, metadata); err != nil {
+					return bundle, errors.Wrapf(err, "bundle load failed on %v", manifestPath)
```

Comment:
I apologize, I don't want to be nitpicky. But lets look at this error message. A possible error that the insert could have would be that the root or non-leaf value are not objects [1] [2]. If this code would hit that issue (which is unlikely), the resulting error would be, for instance: "bundle load failed on non-leaf value must be object". This error doesn't say a lot. You might want to change it to: "metadata insertion to bundle failed at: %v". The goal is to help developers debug this easier. Now, given that this is an unlikely scenario, maybe it would still be OK to not output the whole error message, and just say that there was an error while adding the metadata to the given path. Up to you.

[1] https://github.com/open-policy-agent/opa/blob/master/bundle/bundle.go#L199
[2] https://github.com/open-policy-agent/opa/blob/master/bundle/bundle.go#L224